packages/autorest.python/test/unittests/test_serialization.py

```diff
@@ -2219,9 +2219,14 @@ def __init__(self, name=None, likes_mice=None, dislikes=None, color=None):
         self.assertTrue(animals[2].likes_mice)
 
         message = {"Name": "Didier", "dType": "Animal"}
-        animal = self.d(Animal, message)
-        self.assertIsInstance(animal, Animal)
-        self.assertEqual(animal.name, "Didier")
+        animal1 = self.d(Animal, message)
+        animal2 = Animal.from_dict(message)
+        for animal in [animal1, animal2]:
+            self.assertIsInstance(animal, Animal)
+            self.assertEqual(animal.name, "Didier")
+
+            # deserialize must not change original data
+            self.assertEqual(message["dType"], "Animal")
 
```

Comment:
I made a complicated test which contains nested polymorphic however autorest.python can't handle that scenario and I already created https://github.com/Azure/autorest.python/issues/2804 to track. Now that no users report similar issue, we don't need to consider that complicated scenarios. 
For simple scenarios, changing `pop` to `get` has no influence for deserialization so I think it is ok to review now.